kubernetes/samples/scripts/domain-lifecycle/waitForDomain.sh

```diff
@@ -0,0 +1,598 @@
+  _sortAIImagesUnitTest "aiimages=;c b; bar=;foo; foo=;bar;"   "aiimages=;b,c; bar=;foo; foo=;bar;"
+  _sortAIImagesUnitTest "bar=;foo; foo=;bar; aiimages=; c b ;" "bar=;foo; foo=;bar; aiimages=;b,c;"
+  _sortAIImagesUnitTest "aiimages=;;"                          "aiimages=;;"
+  _sortAIImagesUnitTest "aiimages=; ;"                         "aiimages=;;"
+  _sortAIImagesUnitTest "aiimages=;,,;"                        "aiimages=;;"
+  return 0
+}
+sortAIImagesUnitTest
+
+
+getDomainValue() {
+  # get domain value specified by $2 and put in env var named by $1
+  #   - if get fails, and global EXPECTED_STATE is "Completed", then echo an Error and exit script non-zero
+  #   - example: getDomainValue DOM_VERSION '.spec.introspectVersion'
+  local __retvar=$1
+  local ljpath="{$2}"
+  local attvalue
+  set +e
+  attvalue=$(${KUBERNETES_CLI} -n ${DOMAIN_NAMESPACE} get domain ${DOMAIN_UID} -o=jsonpath="$ljpath" 2>&1)
+  if [ $? -ne 0 ]; then
+    if [ "$EXPECTED_STATE" = "Completed" ]; then
+      trace "Error: Could not obtain '$ljpath' from domain '${DOMAIN_UID}' in namespace '${DOMAIN_NAMESPACE}'. Is your domain resource deployed? Err='$attvalue'"
+      exit 1
+    else
+      # We're waiting for 0 pods - domain might have been deleted, and it doesn't matter what the value is
+      attvalue=''
+    fi
+  fi
+  # echo "DEBUG ${KUBERNETES_CLI} -n ${DOMAIN_NAMESPACE} get domain ${DOMAIN_UID} -o=jsonpath=\"$ljpath\" 2>&1"
+  # echo "DEBUG   = '$attvalue'"
+  eval "$__retvar='$attvalue'"
+  set -e
+}
+
+getClusterValue() {
+  # get cluster value specified by $3 from cluster $1 and put in env var named by $2
+  #   - if get fails, and global EXPECTED_STATE is "Completed", then echo an Error and exit script non-zero
+  #   - example: getClusterValue my-cluster cgen '.metadata.generation'
+  local cname="$1"
+  local __retvar=$2
+  local ljpath="{$3}"
+  local attvalue
+  set +e
+  attvalue=$(${KUBERNETES_CLI} -n ${DOMAIN_NAMESPACE} get cluster ${cname} -o=jsonpath="$ljpath" 2>&1)
+  if [ $? -ne 0 ]; then
+    if [ "$EXPECTED_STATE" = "Completed" ]; then
+      trace "Error: Could not obtain '$ljpath' from cluster '${cname}' in namespace '${DOMAIN_NAMESPACE}'. Is your cluster resource deployed? Err='$attvalue'"
+      exit 1
+    else
+      # We're waiting for 0 pods - cluster might have been deleted, and it doesn't matter what the value is
+      attvalue=''
+    fi
+  fi
+  # echo "DEBUG ${KUBERNETES_CLI} -n ${DOMAIN_NAMESPACE} get cluster ${cname} -o=jsonpath=\"$ljpath\" 2>&1"
+  # echo "DEBUG   = '$attvalue'"
+  eval "$__retvar='$attvalue'"
+  set -e
+}
+
+getDomainAIImages() {
+  # get list of domain auxiliary images (if any) and place result in the env var named by $1
+  #   - if EXPECTED_STATE is "Completed" and get fails, then echo an Error and exit script non-zero
+  #   - result is a sorted comma separated list
+  local attvalue
+  local __retvar=$1
+  set +e
+  attvalue=$(
+    ${KUBERNETES_CLI} \
+      get domain ${DOMAIN_UID} \
```

Comment:
See the previous comment about explicitly getting the v9 version of the domain using ${KUBERNETES_CLI} get domain.v9.weblogic.oracle.

Reply:
Done.